test/replaytests/update-reset-rejected.json

```diff
@@ -0,0 +1,226 @@
+      "eventType": "EVENT_TYPE_WORKFLOW_EXECUTION_STARTED",
+      "taskId": "1050032",
+      "workflowExecutionStartedEventAttributes": {
+        "workflowType": {
+          "name": "CounterWorkflow"
+        },
+        "taskQueue": {
+          "name": "update",
+          "kind": "TASK_QUEUE_KIND_NORMAL"
+        },
+        "workflowExecutionTimeout": "0s",
+        "workflowRunTimeout": "0s",
+        "workflowTaskTimeout": "10s",
+        "originalExecutionRunId": "a178da94-2ac5-4866-b1a5-4c786fa4e29d",
+        "identity": "75101@Quinn-Klassens-MacBook-Pro.local@",
+        "firstExecutionRunId": "a178da94-2ac5-4866-b1a5-4c786fa4e29d",
+        "attempt": 1,
+        "firstWorkflowTaskBackoff": "0s",
+        "header": {},
+        "workflowId": "update-workflow-ID"
+      }
+    },
+    {
+      "eventId": "2",
+      "eventTime": "2024-03-25T22:18:22.830576Z",
+      "eventType": "EVENT_TYPE_WORKFLOW_TASK_SCHEDULED",
+      "taskId": "1050033",
+      "workflowTaskScheduledEventAttributes": {
+        "taskQueue": {
+          "name": "update",
+          "kind": "TASK_QUEUE_KIND_NORMAL"
+        },
+        "startToCloseTimeout": "10s",
+        "attempt": 1
+      }
+    },
+    {
+      "eventId": "3",
+      "eventTime": "2024-03-25T22:18:22.832619Z",
+      "eventType": "EVENT_TYPE_WORKFLOW_TASK_STARTED",
+      "taskId": "1050040",
+      "workflowTaskStartedEventAttributes": {
+        "scheduledEventId": "2",
+        "identity": "74841@Quinn-Klassens-MacBook-Pro.local@",
+        "requestId": "7812cc67-4c7e-46a8-918c-2c744b88aea0",
+        "historySizeBytes": "502"
+      }
+    },
+    {
+      "eventId": "4",
+      "eventTime": "2024-03-25T22:18:26.641635Z",
+      "eventType": "EVENT_TYPE_WORKFLOW_TASK_FAILED",
+      "taskId": "1050069",
+      "workflowTaskFailedEventAttributes": {
+        "scheduledEventId": "2",
+        "startedEventId": "3",
+        "cause": "WORKFLOW_TASK_FAILED_CAUSE_RESET_WORKFLOW",
+        "failure": {
+          "message": "<unknown-user>: test",
+          "resetWorkflowFailureInfo": {}
+        },
+        "identity": "history-service",
+        "baseRunId": "a178da94-2ac5-4866-b1a5-4c786fa4e29d",
+        "newRunId": "3962f036-4c03-4563-95f7-766ca7aa9dd9"
+      }
+    },
+    {
+      "eventId": "5",
+      "eventTime": "2024-03-25T22:18:26.641732Z",
+      "eventType": "EVENT_TYPE_WORKFLOW_EXECUTION_UPDATE_ADMITTED",
+      "taskId": "1050070",
+      "workflowExecutionUpdateAdmittedEventAttributes": {
+        "request": {
+          "meta": {
+            "updateId": "33160fb1-3285-49a6-a655-078dfc30099a",
+            "identity": "75101@Quinn-Klassens-MacBook-Pro.local@"
+          },
+          "input": {
+            "header": {},
+            "name": "fetch_and_add",
+            "args": {
+              "payloads": [
+                {
+                  "metadata": {
+                    "encoding": "anNvbi9wbGFpbg=="
+                  },
+                  "data": "MA=="
```

Comment:
This seems to be `0`, which is non-negative, so why is the update rejected?

Reply:
because there is not update accepted event

Reply:
So, if we were making everything perfectly consistent, we'd make this value in history negative, right?